### PR TITLE
FMS support for chain specific configuration

### PR DESCRIPTION
**Pull request by jkongie**
This commit adds support for specifying FMS configuration on a per chain basis.
Rather than a top level configuration, the user is now asked to choose a chain
they wish to declare to FMS that they support, and attach additional configuration
which will enable FMS to determine which jobs to send to the node, and how to configure
an onchain contract for this specific node.

Additional configuration has been added to support deployment of feeds by FMS.

### CODE REVIEW

File: core/services/feeds/service.go [hunk 1]
```diff
@@ -296,6 +285,82 @@ func (s *service) CountManagers() (int64, error) {
...
+// DeleteChainConfig deletes the chain config by id.
+func (s *service) DeleteChainConfig(id int64) (int64, error) {
+	cfg, err := s.orm.GetChainConfig(id)
+	if err != nil {
+		return 0, errors.Wrap(err, "DeleteChainConfig failed: could not get chain config")
+	}
+
+	_, err = s.orm.DeleteChainConfig(id)
+	if err != nil {
+		return 0, errors.Wrap(err, "CreateChainConfig failed")
```
Comment (PiotrTrzpil) [inline core/services/feeds/service.go:316]: should say: `DeleteChainConfig`

File: core/services/feeds/service.go [hunk 2]
```diff
@@ -296,6 +285,82 @@ func (s *service) CountManagers() (int64, error) {
...
+	}
+
+	mgr, err := s.orm.GetManager(cfg.FeedsManagerID)
+	if err != nil {
+		return 0, errors.Wrap(err, "CreateChainConfig: failed to fetch manager")
```
Comment (PiotrTrzpil) [inline core/services/feeds/service.go:321]: should say: `DeleteChainConfig`

File: core/services/feeds/service.go [hunk 3]
```diff
@@ -296,6 +285,82 @@ func (s *service) CountManagers() (int64, error) {
...
+
+func (s *service) UpdateChainConfig(cfg ChainConfig) (int64, error) {
+	id, err := s.orm.UpdateChainConfig(cfg)
+	if err != nil {
+		return 0, errors.Wrap(err, "UpdateChainConfig failed")
+	}
+
+	ccfg, err := s.orm.GetChainConfig(cfg.ID)
+	if err != nil {
+		return 0, errors.Wrap(err, "DeleteChainConfig failed: could not get chain config")
```
Comment (PiotrTrzpil) [inline core/services/feeds/service.go:354]: should say: `UpdateChainConfig`

**Comment by PiotrTrzpil** [comment]
Looks good! some minor issues

File: core/store/migrate/migrations/0118_create_feeds_manager_chain_config.sql [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+CREATE INDEX idx_feeds_manager_chain_configs_chain_id_chain_type ON feeds_manager_chain_configs(chain_id, chain_type);
+CREATE UNIQUE INDEX idx_feeds_manager_chain_configs_chain_id_chain_type_feeds_manager_id ON feeds_manager_chain_configs(chain_id, chain_type, feeds_manager_id);
+
+-- Remove the old configuration columns
+ALTER TABLE feeds_managers
+DROP CONSTRAINT chk_ocr_bootstrap_peer_multiaddr,
+DROP COLUMN job_types,
+DROP COLUMN is_ocr_bootstrap_peer,
+DROP COLUMN ocr_bootstrap_peer_multiaddr;
```
Comment (PiotrTrzpil) [inline core/store/migrate/migrations/0118_create_feeds_manager_chain_config.sql:25]: no need to migrate them to the new table?

Comment (jkongie) [reply core/store/migrate/migrations/0118_create_feeds_manager_chain_config.sql:25]: No. We can safely drop them as it is not backwards compatible

File: core/web/resolver/mutation.go [hunk 5]
```diff
@@ -135,13 +134,215 @@ func (r *Resolver) DeleteCSAKey(ctx context.Context, args struct {
 	return NewDeleteCSAKeyPayload(key, nil), nil
 }
 
+type createFeedsManagerChainConfigInput struct {
+	FeedsManagerID     string
+	ChainID            string
+	ChainType          string
+	AccountAddr        string
+	AdminAddr          string
```
[the remark this answers is not in the document]
Comment (jkongie) [reply core/web/resolver/mutation.go:142]: I think I use addr in all places, but if not, I can adjust in a future PR

**Review by PiotrTrzpil — APPROVED** [review]

**Pull request merged**